test.py

```diff
@@ -0,0 +1,192 @@
+"""Script to test Flask routes and server"""
+
+from unittest import TestCase
+from server import app
+from flask import session
+import crud
+from model import connect_to_db, db, User, UG_Tweet, Original_Tweet, Author
+import os
+import json 
+
+def example_data():
+    """Create sample data"""
+
+    # In case this is run more than once, empty out existing data
+    User.query.delete()
+    UG_Tweet.query.delete()
+    Original_Tweet.query.delete()
+    Author.query.delete()
+
+    # Add sample users 
+    aurora = crud.create_user('Aurora', 'aurora@libero.it', 'test1')
+    beatrice = crud.create_user('Beatrice', 'beatrice@libero.it', 'test2')
+    claudia = crud.create_user('Claudia', 'claudia@libero.it', 'test3')
+
+    # Add sample authors
+    elon = crud.create_author('Elon Musk', 'elonmusk')
+    kim = crud.create_author('Kim Kardashian West', 'kimkardashian')
+
+    # Add sample Elon test tweets
+    etweet1 = crud.create_original_tweet('Doge Barking at the Moon', elon)
+    etweet2 = crud.create_original_tweet('Tesla AI/Autopilot engineering is awesome! Making excellent progress solving real-world AI.', elon)
+    etweet3 = crud.create_original_tweet('The art In Cyberpunk is incredible', elon)
+
+    # Add sample Kim test tweets
+    ktweet1 = crud.create_original_tweet("You’re always so cute in skims!!!", kim)
+    ktweet2 = crud.create_original_tweet('Hands down - Eye Contour Duo in Medium and the Glossy Lip Balm. The best for a quick natural look', kim)
+    ktweet3 = crud.create_original_tweet("Jeff Leatham or my collaboration w my mom is the perfect gift for Mother’s Day!", kim)
+
+    db.session.add_all([aurora, beatrice, claudia, etweet1, etweet2, etweet3, ktweet1, ktweet2, ktweet3])
+    db.session.commit()
+
+
+class FlaskTestsBasic(TestCase):
+    """Flask server tests"""
+
+    def setUp(self):
+        """Stuff to do before every test."""
+
+        self.client = app.test_client()
+        app.config['TESTING'] = True
+        app.config['LOGIN_DISABLED'] = True
+
+
+    def tearDown(self):
+        """Stuff to do after every test"""
+
+        self.client = None
+        app.config['TESTING'] = False
```

Comment:
Whenever I see code that you can do "for every test" or for "every" something, I would try to make it reusable instead of having to type it out 4-5x. Here, you can do a `class BaseTest(TestCase)` then all your other classes can inherit it like this:

```
class FlaskTestsBasic(BaseTest):
     def setup(self):
           super().setup()
           super().teardown()

Reply:
Awesome, thanks again!